app/src/main/java/com/example/mystore/ui/home/HomeViewModel.kt

```diff
@@ -47,8 +47,10 @@ class HomeViewModel @Inject constructor(
     }
 
     fun filterProducts(newInput: String? = null) {
-        products.update { emptyList() }
-        products.update { allProducts.value.filter { it.price in (priceStartRange..priceEndRange) && it.rating <= rating } }
+        products.value.apply {
+            products.value =
```

Comment:
There's no need for `apply` here if you're just going to use `products.value` once.

Moreover, even if you needed it, `apply` already supplies you with a `this` which is in fact the `products.value` so `this = ....` would be a better usage. As it is right here, the `apply` is not really doing anything.

Tip: If you're wandering what uses does apply have:

1. It infers a `this` within the block so if you do things with the same object on multiple lines you can avoid writing the object variable multiple times but have it just once. Example:
```
list.apply {
  clear()
  addAll(objects)
}
```

which is really a shorthand for

```
list.apply {
  this.clear()
  this.addAll(objects)
}
```

instead of

```
list.clear()
list.addAll(objects)
```
2. It returns the object after the modification which is handy for things like what you've done up here in the same file - `private var selectedCategories = mutableStateListOf<String>().apply { addAll(listOf("Home")) }`. There would be no way to one-line this without an `apply`.